lib/dor/services/suri_service.rb

```diff
@@ -17,8 +17,14 @@ def self.mint_id(quantity = nil)
         ids = resource["identifiers?quantity=#{quantity}"].post('').chomp.split(/\n/).collect { |id| "#{Config.suri.id_namespace}:#{id.strip}" }
       else
         repo = ActiveFedora::Base.respond_to?(:connection_for_pid) ? ActiveFedora::Base.connection_for_pid(0) : ActiveFedora.fedora.connection
-        resp = Nokogiri::XML(repo.next_pid(numPIDs: quantity))
+        resp = Nokogiri::XML(repo.api.next_pid(numPIDs: quantity))
         ids = resp.xpath('/pidList/pid').collect { |node| node.text }
+        # With modernish (circa 2015/6) dependencies, including Nokogiri and
+        # ActiveFedora/Rubydora, `ids` is `[]` above. If that is the case, try
+        # the XPath that works (confirmed with most recent `hydra_etd` work)
+        if ids.empty? && resp.root.namespaces.any?
+          ids = resp.xpath('//xmlns:pid').collect { |node| node.text }
```

Comment:
so this is ... any nodes in the pid namespace?  zounds that seems fraught with ickies.  Might there be particular elements in scope?  Might this xml get big ('//xxx' xpath expressions are expensive, but it doesn't matter with tiny xml).

Reply:
@ndushay The XML is tiny. Fedora 3, which we talk to via Rubydora, allows you to request 1 or more PIDs, and those are returned in an XML blurb that looks like the following, e.g. if you asked for 3 PIDs:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<pidList namespaces="here">
  <pid>druid:123</pid>
  <pid>druid:124</pid>
  <pid>druid:125</pid>
</pidList>
```

You can see the XML generation code in Fedora 3 here: https://github.com/fcrepo3/fcrepo/blob/9c41b09a21a3a2615791fa4c614095a14512940f/fcrepo-server/src/main/java/org/fcrepo/server/rest/DefaultSerializer.java#L65-L84

Reply:
@ndushay I will change the XPath from `//xmlns:pid` to `/xmlns:pidList/xmlns:pid` to more closely mirror the code already in `Dor::SuriService`.